### PR TITLE
Assignment1.2

### LINKED ISSUE #1: Issue- Assignment 1.2
**Exercise**

- Ensure you have SSH access to Tiger GitHub. This allows you to work without having to type in your username/password constantly.

- Create a private repository named mle-training with an empty README.

- Create a GitHub issue for the current assignment and add in the details based on what is requested in this assignment and assign it to yourself. Note the issue-id.

- Create a new branch (enh//) and copy the [code-snippet here](https://drive.google.com/drive/folders/1ebqWUF06mkLRpmV-AjAjPMYgmh0cuUvX) and commit as-is including the README. Use appropriate commit messages as per the guidelines.

- Add a .gitignore file to the repository to avoid committing unnecessary files like pyc files and the CSV data files in your current folder.

- Use Conda to set up a python development environment named mle-dev to run the script. Export the environments file as env.yml and commit it to the repository.

- Update the README file with instructions on how to run the code.

- Create a Pull Request (PR). Make sure the PR follows the expected standards. [Link the PR to the GitHub issue](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) to ensure the issue is closed automatically when the PR is merged.

- Add the facilitators for your session as collaborators on the mle-training repository and request a PR.

- Once the PR is approved, merge the PR to master branch and [give it a tag ](https://git-scm.com/book/en/v2/Git-Basics-Tagging)(named version) v0.1 and push that to the remote.

### PR DESCRIPTION
**Pull request by Dhikshita-macherla**
Closes  #1

- Create a private repository named mle-training
- Created a GitHub issue
- Created a new branch enh/1/issue1.2
- Added a .gitignore file to the repository
- Updated the README file with instructions
- Created a pull request for the changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,22 @@
...
+## Steps performed
+ - We prepare and clean the data. We check and impute for missing values.
+ - Features are generated and the variables are checked for correlation.
+ - Multiple sampling techinuqies are evaluated. The data set is split into train and test.
+ - All the above said modelling techniques are tried and evaluated. The final metric used to evaluate is mean squared error.
+
+## To excute the script
+conda create --name mle-dev <br>
+conda activate mle-dev <br>
+conda env export >environment.yml <br>
```
Comment (abilashravi-ta) [inline README.md:20]: These instructions seem to be wrong..

Ideally, these instructions should tell, anyone who'll be using using your repo, how to 
- clone the repo
- create _conda environment_ from the `envirenment.yml`
- run the script

**Review by abilashravi-ta — CHANGES_REQUESTED** [review]

**Comment by abilashravi-ta** [comment]
@Dhikshita-macherla why is there a _merge conflict_ even for the first PR?

File: README.md (hunk 1 above)
Comment (Dhikshita-macherla) [reply README.md]: I've added the instruction in the README.md file as specified.
I changed the README.md file in main branch after pushing, so i got the merge conflict. Now I've resolved the conflicts. Can you please check and if I need to change anything else.

File: README.md [hunk 2]
```diff
@@ -1 +1,31 @@
...
+<br>
+Creating and Activating the environment<br>
+conda create --name mle-dev <br>
+conda activate mle_dev<br>
+<br>
+Exporting environment file<br>
+conda env export >environment.yml<br>
+<br>
+Executing the python file<br>
+python nostandardcode.py<br>
```
Comment (abilashravi-ta) [inline README.md:29]: @Dhikshita-macherla these instructions doesn't look correct..
Can you have a look at Satya or Sumyami's submission once for an idea.

**Review by abilashravi-ta — CHANGES_REQUESTED** [review]

**Comment by abilashravi-ta** [comment]
Also please correct your [PR description](https://github.com/Dhikshita-macherla/mle-training/pull/2#issue-2183858923)

File: README.md (hunk 2 above)
Comment (Dhikshita-macherla) [reply README.md]: I've changed the above correction. Can you please take a look at it.

File: README.md [hunk 3]
```diff
@@ -1 +1,25 @@
...
+ - Multiple sampling techinuqies are evaluated. The data set is split into train and test.
+ - All the above said modelling techniques are tried and evaluated. The final metric used to evaluate is mean squared error.
+
+## To excute the script
+$git clone git@github.com:Dhikshita-macherla/mle-training.git<br>
+$cd mle-training<br>
+$conda create -f environment.yml<br>
+$conda create --name mle-dev<br>
+$conda activate mle_dev<br>
+$conda env export >environment.yml<br>
```
Comment (abilashravi-ta) [inline README.md:23]: This **Line 23** need not be part of the instructions to run the script..

**Review by abilashravi-ta — APPROVED** [review]
LGTM. You can _merge_ it to **main**.

@Dhikshita-macherla please make all the changes on _VSCode_ or _Terminal_ in local repository and then _push_ your changes to remote.
Don't directly commit on _GitHub_.

File: README.md (hunk 3 above)
Comment (Dhikshita-macherla) [reply README.md:23]: Okay, thank you Abilash.

**Pull request merged**